scadnano/scadnano.py

```diff
@@ -6833,15 +6834,32 @@ def _oxdna_get_helix_vectors(design: Design, helix: Helix) -> Tuple[_OxdnaVector
     grid = group.grid
     geometry = design.geometry
 
-    forward = _OxdnaVector(0, 0, 1)
-    normal = _OxdnaVector(0, -1, 0)
-
-    forward = forward.rotate(design.yaw_of_helix(helix), normal)
-    forward = forward.rotate(-design.pitch_of_helix(helix), _OxdnaVector(1, 0, 0))
-    normal = normal.rotate(-design.pitch_of_helix(helix), _OxdnaVector(1, 0, 0))
-
-    normal = normal.rotate(-helix.roll, forward)
+    # principal axes for computing rotation
+    # see https://en.wikipedia.org/wiki/Aircraft_principal_axes
+    yaw_axis = _OxdnaVector(0, 1, 0)
+    pitch_axis = _OxdnaVector(0, 0, 1)
+    roll_axis = _OxdnaVector(1, 0, 0)
 
+    # we apply rotations in the order yaw, pitch, and then roll
+    # the _OxdnaVector.rotate function applies ccw rotation so angle needs to be negated
+    
+    # first the yaw rotation
+    pitch_axis = pitch_axis.rotate(-design.yaw_of_helix(helix), yaw_axis)
+    roll_axis = roll_axis.rotate(-design.yaw_of_helix(helix), yaw_axis)
+
+    # then the pitch rotation
+    yaw_axis = yaw_axis.rotate(-design.pitch_of_helix(helix), pitch_axis)
+    roll_axis = roll_axis.rotate(-design.pitch_of_helix(helix), pitch_axis)
+
+    # then the roll rotation
+    yaw_axis = yaw_axis.rotate(-design.roll_of_helix(helix), roll_axis)
+    pitch_axis = pitch_axis.rotate(-design.roll_of_helix(helix), roll_axis)
```

Comment:
This code seems suspicious to me. The variable `pitch_axis` is not used after this point, so if assigning to it is supposed to have some effect, it won't.

Also, it isn't what I was expecting. The example shown in https://github.com/UC-Davis-molecular-computing/scadnano/issues/667, which looks like this in scadnano:

![image](https://user-images.githubusercontent.com/19274365/140983235-3b920747-32d5-4467-a077-c57f3d099bf6.png)

now displays in oxDNA like this:

![image](https://user-images.githubusercontent.com/19274365/140982906-f2226d47-d99a-4d4e-9df5-5e42cb843ca0.png)

It should look like this (but with the blue axis pointing right and the green axis pointing down):

![image](https://user-images.githubusercontent.com/19274365/140983139-776c66a2-e42c-48cc-8a09-2f0ab47ec234.png)